state/protocol/badger/mutator.go

```diff
@@ -293,22 +269,15 @@ func (m *ParticipantState) Extend(ctx context.Context, candidateProposal *flow.P
 		return err
 	}
 
-<<<<<<< HEAD
-	// The following function rejects the input block with an [state.OutdatedExtensionError] if and only if
-	// the block is orphaned or already finalized. If the block was to be finalized already, it would have been
-	// detected as already processed by the check above. Hence, `candidate` being orphaned is the only
-	// possible case to receive an [state.OutdatedExtensionError] here.
-	err = m.checkOutdatedExtension(candidate.Header)
-=======
-	// check if the block header is a valid extension of parent block
-	err = m.headerExtend(ctx, candidateProposal, nil, deferredDbOps)
 	if err != nil {
 		return fmt.Errorf("header not compliant with chain state: %w", err)
 	}
 
-	// check if the block header is a valid extension of the finalized state
-	err = m.checkOutdatedExtension(candidate.HeaderBody)
->>>>>>> master
+	// The following function rejects the input block with an [state.OutdatedExtensionError] if and only if
+	// the block is orphaned or already finalized. If the block was to be finalized already, it would have been
+	// detected as already processed by the check above. Hence, `candidate` being orphaned is the only
+	// possible case to receive an [state.OutdatedExtensionError] here.
+	err = m.checkOutdatedExtension(candidate.ToHeader())
```

Comment:
Why we have reordered `checkOutdatedExtension` and `headerExtend`? To me it makes sense to keep it as it was before since `headerExtend` performs some validity checks on the header body on the other hand `checkOutdatedExtension` performs a potentially long operation which is querying blocks is an unfinalized fork which could grow quite large in some scenarios.

Reply:
Good catch 👍 . Fixed this [1a6d796](https://github.com/onflow/flow-go/pull/7795/commits/1a6d79632ec99db3d4b316b4b45cc26e855a1e5d)